stablehlo/reference/Ops.cpp

```diff
@@ -54,6 +54,20 @@ Tensor evalAndOp(const Tensor &lhs, const Tensor &rhs, Type resultType) {
   return result;
 }
 
+Tensor evalBroadcastInDimOp(const Tensor &operand,
+                            ArrayRef<int64_t> broadcastDimensions,
+                            Type resultType) {
+  Tensor result(resultType);
+  auto operandShape = operand.getType().getShape();
+  for (auto it = result.index_begin(); it != result.index_end(); ++it) {
+    SmallVector<int64_t> inIndices;
+    for (auto [inDim, outDim] : llvm::enumerate(broadcastDimensions))
+      inIndices.push_back(operandShape[inDim] == 1 ? 0 : (*it)[outDim]);
+    result.set(*it, operand.get(inIndices));
```

Comment:
Feedback on names:
  1) `it` => `resultIt` for consistency with other eval functions.
  2) `inIndices` => `operandIdx` for consistency with other eval functions.
  3) `inDim` => `operandDim` to follow ODS names.
  4) `outDim` => `resultDim` to follow ODS names.